### PR TITLE
Handle privatecommand messages that pass no data

### PR DESCRIPTION
**Pull request by justinmc**
Currently, if a `TextInputClient.performPrivateCommand` platform message is sent that has no `data` key, the framework will not know how to handle it and will throw an error.  This PR properly handles this situation and eliminates the error.

The engine purposely sends a map with no `data` key in situations where the data map isn't used ([code](https://github.com/flutter/engine/blob/b48eeca17b7cae6506af1136e85116ccf0d7c9e3/shell/platform/android/io/flutter/embedding/engine/systemchannels/TextInputChannel.java#L355)).

This PR allows receiving a null value and passing it through to onAppPrivateCommand.

### CODE REVIEW

File: packages/flutter/lib/src/services/text_input.dart [hunk 1]
```diff
@@ -1837,7 +1837,9 @@ class TextInput {
         final Map<String, dynamic> firstArg = args[1] as Map<String, dynamic>;
         _currentConnection!._client.performPrivateCommand(
           firstArg['action'] as String,
-          firstArg['data'] as Map<String, dynamic>,
+          firstArg['data'] == null
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/services/text_input.dart:1840]: nit: `as Map<String, dynamic>?`

File: packages/flutter/lib/src/services/text_input.dart [hunk 2]
```diff
@@ -1134,7 +1134,7 @@ mixin TextInputClient {
   ///   * [sendAppPrivateCommand](https://developer.android.com/reference/android/view/inputmethod/InputMethodManager#sendAppPrivateCommand),
   ///     which is the Android documentation for sendAppPrivateCommand, used to
   ///     send a command to the input method.
-  void performPrivateCommand(String action, Map<String, dynamic> data);
+  void performPrivateCommand(String action, Map<String, dynamic>? data);
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/services/text_input.dart:1137]: This breaks internal tests I think?

**Review by LongCatIsLooong — APPROVED** [review]

File: packages/flutter/lib/src/services/text_input.dart (hunk 1 above)
Comment (justinmc) [reply packages/flutter/lib/src/services/text_input.dart:1840]: Oh duh, that's awesome that we can do that 👍

File: packages/flutter/lib/src/services/text_input.dart (hunk 2 above)
Comment (justinmc) [reply packages/flutter/lib/src/services/text_input.dart:1137]: I'll try running the Google tests on this PR when it goes green.  I thought about just creating an empty map if it's null and keeping this parameter the same non-nullable, but it felt dirty.  I'll consider just migrating any breakages myself if it's not too crazy.

**Comment by LongCatIsLooong** [comment]
I LGTM'd earlier, google testing is now finished.

**Comment by justinmc** [comment]
Ah sorry you're right.  Well it looks like there are tons of breaking changes from that.  I should have assumed lots of people are implementing TextInputClient.  I will use an empty Map to avoid the breaking change.

**Comment by absar** [comment]
@LongCatIsLooong given the small size of the fix can it be cherry picked into stable or beta, as it's flooding crashlytics on old Android devices with each tap in a text field, related issue https://github.com/flutter/flutter/issues/113973

**Comment by absar** [comment]
@justinmc I don't see this in beta branch, it should be cherry-picked there as well

**Comment by justinmc** [comment]
@absar Now it looks like it's released on beta.

https://github.com/flutter/flutter/blob/75927305ff855f76a9ef704f9b4a86fa2fce7292/packages/flutter/lib/src/services/text_input.dart#L1880-L1882

**Pull request merged**